cloudwatch_logger/src/log_node_param_helper.cpp

```diff
@@ -0,0 +1,202 @@
+}
+
+Aws::AwsError ReadMinLogVerbosity(
+  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
+  int8_t & min_log_verbosity)
+{
+  min_log_verbosity = kNodeMinLogVerbosityDefaultValue;
+
+  std::string specified_verbosity;
+  Aws::AwsError ret =
+    parameter_reader->ReadStdString(kNodeParamMinLogVerbosityKey, specified_verbosity);
+  switch (ret)
+  {
+    case AwsError::AWS_ERR_NOT_FOUND:
+      AWS_LOGSTREAM_WARN(__func__, "Log verbosity configuration not found, setting to default value: "
+                                   << kNodeMinLogVerbosityDefaultValue);
+      break;
+    case AwsError::AWS_ERR_OK:
+      if ("DEBUG" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::DEBUG;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to DEBUG.");
+      } else if ("INFO" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::INFO;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to INFO.");
+      } else if ("WARN" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::WARN;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to WARN.");
+      } else if ("ERROR" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::ERROR;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to ERROR.");
+      } else if ("FATAL" == specified_verbosity) {
+        min_log_verbosity = rosgraph_msgs::Log::FATAL;
+        AWS_LOG_INFO(__func__, "Log verbosity is set to FATAL.");
+      } else {
+        ret = AwsError::AWS_ERR_PARAM;
+        AWS_LOGSTREAM_WARN(__func__,
+                          "Log verbosity configuration not valid, setting to default value: "
+                           << kNodeMinLogVerbosityDefaultValue);
+      }
+      break;
+    default:
+      AWS_LOGSTREAM_ERROR(__func__, 
+        "Error " << ret << " retrieving log verbosity configuration, setting to default value: "
+        << kNodeMinLogVerbosityDefaultValue);
+  }
+  return ret;
+}
+
+Aws::AwsError ReadSubscriberList(
+  const bool subscribe_to_rosout,
+  std::shared_ptr<Aws::Client::ParameterReaderInterface> parameter_reader,
+  boost::function<void(const rosgraph_msgs::Log::ConstPtr &)> callback,
+  ros::NodeHandle & nh,
+  std::vector<ros::Subscriber> & subscriptions)
+{
+  std::vector<std::string> topics;
+  Aws::AwsError ret = parameter_reader->ReadList(kNodeParamLogTopicsListKey, topics);
+
+  for (const std::string& topic : topics)
+  {
+    ros::Subscriber sub = nh.subscribe(topic, kNodeSubQueueSize, callback);
+    AWS_LOGSTREAM_INFO(__func__, "Subscribing to topic: " << topic);
+    subscriptions.push_back(sub);
+  }
+  if (subscribe_to_rosout) {
+    ros::Subscriber sub = nh.subscribe(kNodeRosoutAggregatedTopicName, kNodeSubQueueSize, callback);
+    AWS_LOG_INFO(__func__, "Subscribing to rosout_agg");
+    subscriptions.push_back(sub);
+  }
+  return ret;
+}
+
+
+}  // namespace Utils
+}  // namespace CloudWatchLogs
+}  // namespace Aws
```

Comment:
Add new line at the end of each of these files.

Reply:
done